mobile/lib/components/ramowka.dart

```diff
@@ -0,0 +1,398 @@
+            .map(
+              (ramowkaInfo) => _RamowkaListItem(
+                info: ramowkaInfo,
+                rowHeight: widget.rowHeight,
+              ),
+            )
+            .toList(),
+      );
+    } else {
+      return _RamowkaNoData(height: height);
+    }
+  }
+}
+
+/// Represents a single [RamowkaList] entry.
+class _RamowkaListItem extends StatelessWidget {
+  const _RamowkaListItem({
+    required this.info,
+    required this.rowHeight,
+  });
+
+  final RamowkaInfo info;
+  final double rowHeight;
+
+  /// Calculates aspect ratio from screen width and this
+  /// widget's single row height used for max text length.
+  double aspectRatio(BuildContext context) =>
+      MediaQuery.of(context).size.width / rowHeight;
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.only(left: 3),
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        children: [
+          Material(
+            color: Colors.transparent,
+            child: InkWell(
+              // TODO: Navigation to this Audycja
+              onTap: () {},
+              splashColor: context.colors.highlightGreen.withOpacity(0.3),
+              radius: 25,
+              child: AspectRatio(
+                aspectRatio: aspectRatio(context),
+                child: Text(
+                  info.title,
+                  style: context.textStyles.textSmall,
+                  overflow: TextOverflow.ellipsis,
+                  maxLines: 1,
+                ),
+              ),
+            ),
+          ),
+          Text(
+            info.startTime,
+            style: context.textStyles.textSmall,
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+/// Empty variant of [RamowkaList]
+class _RamowkaNoData extends StatelessWidget {
+  const _RamowkaNoData({required this.height});
+
+  final double height;
+
+  @override
+  Widget build(BuildContext context) {
+    return RaListWidget(
+      rows: 1,
+      rowHeight: height,
+      items: [
+        Center(
+          child: Text(
+            'Wystąpił błąd podczas pobierania danych',
+            style: context.textStyles.textSmall,
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+/// Variant of the [RamowkaWidget] containing
+/// a waiting animation
+class _RamowkaWaiting extends StatelessWidget {
+  const _RamowkaWaiting({required this.height});
+
+  final double height;
+
+  @override
+  Widget build(BuildContext context) {
+    return RaListWidget(
+      rows: 1,
+      rowHeight: height,
+      items: [
+        Center(
+          child: SizedBox(
+            width: 30,
+            height: 30,
+            child: CircularProgressIndicator(
+              color: context.colors.highlightGreen,
+              backgroundColor: context.colors.backgroundDark,
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+/// Information about single [RamowkaList] entry.
+class RamowkaInfo {
```

Comment:
extract classes containing RamowkaItems to separate files in one Ramowka folder [Solid](https://en.wikipedia.org/wiki/Single-responsibility_principle)

Reply:
Good suggestion